packages/cache/src/cache.ts

```diff
@@ -181,18 +171,45 @@ export async function saveCache(
     if (core.isDebug()) {
       await listTar(archivePath, compressionMethod)
     }
-
     const fileSizeLimit = 10 * 1024 * 1024 * 1024 // 10GB per repo limit
     const archiveFileSize = utils.getArchiveFileSizeInBytes(archivePath)
     core.debug(`File Size: ${archiveFileSize}`)
-    if (archiveFileSize > fileSizeLimit) {
+    // For GHES, this check will take place in ReserveCache API with enterprise file size limit
+    if (archiveFileSize > fileSizeLimit && !utils.isGhes()) {
       throw new Error(
         `Cache size of ~${Math.round(
           archiveFileSize / (1024 * 1024)
         )} MB (${archiveFileSize} B) is over the 10GB limit, not saving cache.`
       )
     }
 
+    const cacheSize = archiveFileSize
+    core.debug('Reserving Cache')
+    const reserveCacheResponse = await cacheHttpClient.reserveCache(
+      key,
+      paths,
+      {
+        compressionMethod,
+        cacheSize
+      }
+    )
+
+    if (reserveCacheResponse?.result?.cacheId) {
+      cacheId = reserveCacheResponse?.result?.cacheId
+    } else if (reserveCacheResponse?.statusCode === 400) {
+      throw new ReserveCacheError(
+        reserveCacheResponse?.error?.message ??
+          `Cache size of ~${Math.round(
```

Comment:
Should the server return a message and we show that here instead? Just assuming every 400 is a Bad Request may not scale well if we return 400's for other issues.

Reply:
Wont `reserveCacheResponse?.error?.message ??` handle that? We are first checking if message is present in the response or not. If not only then we fallback to the default value.